app/frontend/courses/curricula/components/CoursesCurriculum__SubmissionCommentShow.res

```diff
@@ -0,0 +1,236 @@
+%%raw(`import "./CoursesCurriculum__SubmissionCommentShow.css";`)
+let str = React.string
+let tr = I18n.t(~scope="components.CoursesCurriculum__SubmissionCommentShow")
+
+open CoursesCurriculum__Types
+
+module HideSubmissionCommentMutation = %graphql(`
+   mutation HideSubmissionCommentMutation($submissionCommentId: String!, $hide: Boolean!) {
+     hideSubmissionComment(submissionCommentId: $submissionCommentId, hide: $hide ) {
+      success
+     }
+   }
+   `)
+
+module ArchiveSubmissionCommentMutation = %graphql(`
+   mutation ArchiveSubmissionCommentMutation($submissionCommentId: String!) {
+     archiveSubmissionComment(submissionCommentId: $submissionCommentId ) {
+       success
+     }
+   }
+   `)
+
+let hideComment = (submissionCommentId, hide, setCommentHidden, event) => {
+  ReactEvent.Mouse.preventDefault(event)
+  HideSubmissionCommentMutation.make({submissionCommentId, hide})
+  |> Js.Promise.then_(response => {
+    if response["hideSubmissionComment"]["success"] {
+      setCommentHidden(_ => hide)
+    }
+    Js.Promise.resolve()
+  })
+  |> ignore
+}
+
+let archiveComment = (submissionCommentId, setCommentArchived, setShowConfirmDelete, event) => {
+  ReactEvent.Mouse.preventDefault(event)
+  setShowConfirmDelete(_ => false)
+  ArchiveSubmissionCommentMutation.make({submissionCommentId: submissionCommentId})
+  |> Js.Promise.then_(response => {
+    if response["archiveSubmissionComment"]["success"] {
+      setCommentArchived(_ => true)
+    }
+    Js.Promise.resolve()
+  })
+  |> ignore
+}
+
+let updateShowConfirmDelete = (setShowConfirmDelete, showConfirmDelete, event) => {
+  ReactEvent.Mouse.preventDefault(event)
+  setShowConfirmDelete(_ => showConfirmDelete)
+}
+
+@react.component
+let make = (~currentUser, ~comment) => {
+  let (commentHidden, setCommentHidden) = React.useState(() =>
+    Belt.Option.isSome(comment->Comment.hiddenAt)
+  )
+  let (commentArchived, setCommentArchived) = React.useState(() => false)
+  let (showConfirmDelete, setShowConfirmDelete) = React.useState(() => false)
+
+  let isModerator = currentUser->CurrentUser.isModerator
+  let userName = comment->Comment.user->User.name
+
+  let commentDisplay =
+    <div
```

Comment:
@pupilfirst/designers I get the feeling that these comments could have had _tighter_ design. What I'm thinking of is StackOverflow's comments underneath their posts - it packs these features (and more?) into a far more dense view.